tests/test_database_service.py

```diff
@@ -204,6 +208,35 @@ def test_get_dblp_authors(self) -> None:
             TestBrowseDatabaseService.database_service.get_dblp_authors(
                 test_paper_id), [])
 
+    @mock.patch('browse.services.database.models.db.session.query')
+    def test_error_conditions(self, mock_query)->None:
+        mock_query.side_effect = NoResultFound
+        self.assertEqual(
+            TestBrowseDatabaseService.database_service.get_institution('10.0.0.1'), None)
+        self.assertEqual([],
+            TestBrowseDatabaseService.database_service.get_all_trackback_pings())
+        self.assertListEqual(
+            TestBrowseDatabaseService.database_service.get_trackback_pings('0704.0361'), [])
+        self.assertEqual(
+            TestBrowseDatabaseService.database_service.count_trackback_pings('0704.0361'), 0)
+        self.assertEqual(
+            TestBrowseDatabaseService.database_service.count_all_trackback_pings(), 0)
+        self.assertEqual(
+            TestBrowseDatabaseService.database_service.has_sciencewise_ping('0704.0361'), False)
+        self.assertEqual(
+            TestBrowseDatabaseService.database_service.get_dblp_listing_path('0704.0361'), None)
+        self.assertEqual(
+            TestBrowseDatabaseService.database_service.get_dblp_authors('0704.0361'), [])
+        mock_query.side_effect = SQLAlchemyError
+        self.assertRaises(SQLAlchemyError,
```

Comment:
👍 Nice, I didn't see this before! Still curious about dropping the SQLAlchemyError handling though :)

Reply:
Each path through exception handling code requires additional testing. So letting as many types of exceptions go to arxiv-base 500 as possible makes it so we can test fewer types of exceptions. Almost all will be handled uniformly by arxiv-base's 500 and we know we have that tested.